tokio-process/tests/issue_42.rs

```diff
@@ -16,20 +16,19 @@ fn run_test() {
     let finished_clone = finished.clone();
 
     thread::spawn(move || {
-        let _ = stream::iter_ok(0..2)
-            .map(|i| {
+        let mut futures = stream::FuturesOrdered::new();
+        for i in 0..2 {
+            futures.push(
                 Command::new("echo")
                     .arg(format!("I am spawned process #{}", i))
                     .stdin(Stdio::null())
                     .stdout(Stdio::null())
                     .stderr(Stdio::null())
                     .spawn_async()
-                    .into_future()
-                    .flatten()
-            })
-            .buffered(2)
-            .collect()
-            .wait();
+                    .unwrap(),
+            )
+        }
+        futures::executor::block_on_stream(futures);
```

Comment:
Blocking: According to the docs this function converts the stream to a blocking iterator, which is still lazy. My hunch is that we aren't actually invoking any futures so I think this test is currently broken.

I think we should stick to using `.collect()` on the stream and using `tokio::block_on` to consume it fully

Reply:
I reimplemented this using `collect`. PTAL